### PR TITLE
adding cpuid for lpc1768 cpu, enable periph_cpuid in board mbed_lpc1768

### PR DESCRIPTION
**Pull request by nqd**
Adding cpu id for lpc1768 cpu, inspiring from #1511.

### CODE REVIEW

File: cpu/lpc1768/periph/cpuid.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    void (*iap)(uint32_t[], uint32_t[]);
+    /* Set the function pointer */
+    iap = (void (*)(uint32_t[], uint32_t[])) IAP_ADDRESS;
+    /* Read UUID */
+    iap(&command, result);
+
+    if (result[0] == 0)
+        memcpy(id, &result[1], CPUID_ID_LEN);
+    else
+        memset(id, 0xFF, CPUID_ID_LEN);
```
Comment (OlegHahm) [inline cpu/lpc1768/periph/cpuid.c]: Can you add parentheses around the if and else blocks, please?

File: cpu/lpc1768/periph/cpuid.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * @author      Paul RATHGEB <paul.rathgeb@skynet.be>
+ * @author      Dinh Nguyen <nqdinhddt@gmail.com>
+ */
+
+#include <string.h>
+#include "periph/cpuid.h"
+
+/* IAP base address */
+#define IAP_ADDRESS 0x1FFF1FF1
```
Comment (OlegHahm) [inline cpu/lpc1768/periph/cpuid.c]: Please add parens here, too.

**Comment by OlegHahm** [comment]
Sorry, this one slipped through.

**Comment by nqd** [comment]
hello @OlegHahm, could you please check for code convention and result of the updated 2 commit?

**Comment by OlegHahm** [comment]
Looks (almost) good. If you can add the missing parentheses for the `IAP_ADDRESS` define, we should be ready to merge.

**Comment by nqd** [comment]
oh, sorry (should look at doxygen reqr more). Good luck to code reviewer :).

**Comment by OlegHahm** [comment]
Ah, sorry, I meant:
```C
/**
 * @name IAP base address
 */
#define IAP_ADDRESS     (0x1FFF1FF1)
```
i.e. parentheses around the value.

**Comment by nqd** [comment]
should be good with last --force push

**Comment by OlegHahm** [comment]
```
xtimer_now64_continuity:mbed_lpc1768:

Building application "xtimer_now64_continuity" for "mbed_lpc1768" with MCU "lpc1768".

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c: In function 'cpuid_get':

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:31:5: error: unknown type name 'uint32_t'

     uint32_t result[5];

     ^

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:33:5: error: unknown type name 'uint32_t'

     uint32_t command = 58;

     ^

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:35:17: error: unknown type name 'uint32_t'

     void (*iap)(uint32_t[], uint32_t[]);

                 ^

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:35:29: error: unknown type name 'uint32_t'

     void (*iap)(uint32_t[], uint32_t[]);

                             ^

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:37:5: error: 'iap' undeclared (first use in this function)

     iap = (void (*)(uint32_t[], uint32_t[])) IAP_ADDRESS;

     ^

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:37:5: note: each undeclared identifier is reported only once for each function it appears in

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:37:21: error: unknown type name 'uint32_t'

     iap = (void (*)(uint32_t[], uint32_t[])) IAP_ADDRESS;

                     ^

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:37:33: error: unknown type name 'uint32_t'

     iap = (void (*)(uint32_t[], uint32_t[])) IAP_ADDRESS;

                                 ^

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:27:25: error: expected ';' before numeric constant

 #define IAP_ADDRESS     0x1FFF1FF1

                         ^

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:37:46: note: in expansion of macro 'IAP_ADDRESS'

     iap = (void (*)(uint32_t[], uint32_t[])) IAP_ADDRESS;

                                              ^

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:39:5: error: implicit declaration of function 'iap' [-Werror=implicit-function-declaration]

     iap(&command, result);

     ^

/home/travis/build/RIOT-OS/RIOT/cpu/lpc1768/periph/cpuid.c:42:32: error: 'CPUID_ID_LEN' undeclared (first use in this function)

         memcpy(id, &result[1], CPUID_ID_LEN);

                                ^

cc1: all warnings being treated as errors
```

**Comment by OlegHahm** [comment]
@nqd, please add the `include <stdint.h>` and you should be fine.

**Comment by nqd** [comment]
I cannot repeat the travis error in my laptop and desktop, since `stdint` is included in the chain already.

```
$cat bin/mbed_lpc1768/periph/cpuid.d 
 /home/nqdinh/workspaces/RIOT/examples/hello-world/bin/mbed_lpc1768/periph/cpuid.o: \
...
/home/nqdinh/workspaces/RIOT/cpu/cortexm_common/include/core_cm3.h \
/usr/include/newlib/stdint.h \
...
```

ANW, just added this header directly and see travis outcome. I don't have time to invest travis configure to know why; hence very nice if you could give me brief explain :).

**Comment by OlegHahm** [comment]
Another error:
```
cpuid.c: In function 'cpuid_get':
cpuid.c:43:32: error: 'CPUID_ID_LEN' undeclared (first use in this function)
         memcpy(id, &result[1], CPUID_ID_LEN);
```

**Comment by nqd** [comment]
@OlegHahm I will fix this tonight. When travis and riot ci green, i will ping you.

**Comment by nqd** [comment]
Stupid me. Forgot to rebase master.

@OlegHahm should be ok now :).

**Comment by haukepetersen** [comment]
just tested - works like a charme. Travis is happy, so ACK and go

**Pull request merged**